orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -43,19 +43,56 @@ public TrafficGuard(OortHelper oortHelper, Front50Service front50Service) {
     this.front50Service = front50Service;
   }
 
+  public void verifyInstanceTermination(List<String> instanceIds, String account, Location location, String cloudProvider, String operationDescriptor) {
+    Map<String, List<String>> instancesPerServerGroup = new HashMap<>();
+    instanceIds.forEach(instanceId -> {
+
+      String resolvedServerGroupName = resolveServerGroupNameForInstance(instanceId, account, location.getValue(), cloudProvider);
+      if (resolvedServerGroupName != null) {
+        instancesPerServerGroup.computeIfAbsent(resolvedServerGroupName, serverGroup -> new ArrayList<>()).add(instanceId);
+      }
+    });
+
+    instancesPerServerGroup.entrySet().forEach(entry -> {
+      String serverGroupName = entry.getKey();
+      Names names = Names.parseName(serverGroupName);
+      if (hasDisableLock(names.getCluster(), account, location)) {
+        Optional<TargetServerGroup> targetServerGroup = oortHelper.getTargetServerGroup(account, serverGroupName, location.getValue(), cloudProvider);
+
+        targetServerGroup.ifPresent(serverGroup -> {
+          long otherActiveInstances = serverGroup.getInstances().stream().filter(i -> "Up".equals(i.get("healthState")) && !entry.getValue().contains(i.get("name"))).count();
```

Comment:
Good point, it'll be `Unknown`. I don't have a good answer for this, beyond traffic guards should only be configured for applications that report instance health. We could work off the "interestingHealthProviders" field on the application config, potentially, but it's possible that might get us into trouble...